core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2797,63 +2590,65 @@ object LogMetricNames {
   }
 }
 
-sealed trait SegmentDeletionReason {
-  def logReason(log: Log, toDelete: List[LogSegment]): Unit
+abstract class SegmentDeletionReason {
+  def logReason(logger: Logging, toDelete: List[LogSegment]): Unit = {}
```

Comment:
Hmm, this method is being used, but no subclass implements it. Are we losing logging because of this?

Reply:
Great catch. After an investigation, I realized this change is no longer required and I've reverted it now.